src/main/resources/avro/sequenceAnnotations.avdl

```diff
@@ -98,9 +103,9 @@ protocol SequenceAnnotations {
     }> attributes;
 
     /**
-    Identifiers of children nodes
+    Identifier for the parent annotation, if any.
     */
-    array<string> childrenIds;
+    string parentId;
```

Comment:
CH Albach <notifications@github.com> writes:
> multiple parents. Some workarounds for this are:
> 
>   • create two transcript resources instead of one; link them together or have
>     some ID which indicates that they originated from the same source data.
>
>   • put other parent IDs into a second-class tag in the attributes field.

I want to avoid special cases if at all possible.  They break
lots of stuff (e.g. human PAR region) because people ignore
them.  However, I think we can just do away with multiple parent
representation.

The only use case I have found is exons shared between
transcripts.  This is an alternate representation, not a special
case, and it's not used much, if at all, in the vertebrate
world.  The reason for this is the exon annotations can differ
between transcripts.  A CDS in one transcript might be UTR in
another.

So if we define the SeqAnnRegion objects to represent a simple,
hierarchical set of annotations, all other relations ships are
expressed by other links in the attributes.

Daniel's interesting cases:
> Good point, a transcript can be assigned multiple parent genes (e.g. fusion
> transcript, trans-spliced, etc)

Cancer fusions transcripts would actually be two separate gene
annotations, as they are not the original gene.  These then get
linked.  I have to re-remember how trans-splicing is coded in GFF3.

> A few of my reasons as to why I prefer parentId over childIds:
> 
>   • Traversal of the tree is cleanest via Search(byParentId). The alternative
>     is to take the list of childIds and perform a batchGet call with those IDs
>     (which we would have to add), or N getSeqAnnRegion calls. If we have a
>     search by parentId, it is natural for the records to have a parentId field
>     and the childIds field has no utility for traversal.

A "return all my children query" is very important, however this
doesn't need to be represented in the data structure. Really,
the id in the data structure is for linking records together
once you get them back.  This should be independent of the
querys. One could have queries that support `give me my parent',
'give me my children'.  But you have to have at least one ID
to link the results.


>   • If the number of children for a given annotation becomes large, you wind up
>     with an unwieldy childIds array.

At first this didn't move me, because most are like less than
10-20, although there are a few cases in the 100s.  Computers
can handle that.  I then realized this is an issue with
constructing URLs, which is compelling.

OK, I am sold, I will switch to a single parent id and start
documenting how this differs from GFF3.

Mark